fastify.d.ts

```diff
@@ -242,6 +242,13 @@ declare namespace fastify {
      */
     options(url: string, handler: RequestHandler<HttpRequest, HttpResponse>): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
 
+    /**
+     * Starts the server on the given port after all the plugins are loaded,
+     * internally waits for the .ready() event. Using promise.
+     */
+    listen(port: number): Promise<http.Server>
+    listen(port: number, hostname?: string): Promise<http.Server>
```

Comment:
I don’t think we are returning anything here. Is <http.Server> needed at all?

Reply:
I see callbacks' usage return http.Server so, I added it.

Reply:
Hmm, probably not needed. Maybe should return just `Promise<void>`.

Reply:
listen returns undefined everywhere. So you might as well fix it.